### PR TITLE
Added proper responses for http code 502, 503 and 504

### PR DESCRIPTION
**Pull request by tboerger**

### CODE REVIEW

**Comment by rsalevsky** [comment]
+1

File: lib/crowbar/client/request/base.rb [hunk 1]
```diff
@@ -60,15 +60,24 @@ def process
...
-            raise InternalServerError,
-              "An internal error occured"
           when 401
             raise NotAuthorizedError,
               "User is not authorized"
           when 403
             raise NotAuthorizedError,
               "User access is forbidden"
+          when 500
+            raise InternalServerError,
```
Comment (MaximilianMeister) [inline lib/crowbar/client/request/base.rb]: why do you need to implement your own empty HTTP error classes when you have the ones from `net/http`?

Comment (tboerger) [inline lib/crowbar/client/request/base.rb]: > why do you need to implement your own empty HTTP error classes when you have the ones from net/http?

I have even not taken any look at it. From httparty i just get error codes.

**Comment by vuntz** [comment]
Travis is failing here.

**Comment by tboerger** [comment]
@vuntz this was related to rubocop because of too much complexity. Should be fixed now.

**Comment by vuntz** [comment]
+1

**Pull request merged**